### PR TITLE
Adjust e2e transfer and governance tests to match new fee distribution and eliminate ProposerFraction

### PR DESCRIPTION
**Pull request by nategraf**
### Description

Fee distribution updated in https://github.com/celo-org/celo-blockchain/pull/563 requires updates to the e2e transfer test. This PR provides those changes.

### Tested

Tested with e2e and existing unit tests.

### Other changes

* Only restart the geth nodes between different sync options to avoid a fragile test condition in which we only test transfers from a fresh state.
* Shorten the delay on waiting for the sync node to update.
* Check gas usage on test transfers, because the tests may erroneous pass or fail a test condition if the gas usage changes without the test being updated.

### Related issues

- Related to #540

### Backwards compatibility

Introduces a breaking change to contract kit by removing proposer fraction. Test changes are compatible with blockchain clients which include https://github.com/celo-org/celo-blockchain/pull/563

Depends on https://github.com/celo-org/celo-blockchain/pull/563

### CODE REVIEW

**Comment by nategraf** [comment]
Marking `do not merge` until celo-org/celo-blockchain#563 is merged

File: .circleci/config.yml
Comment (asaj) [inline .circleci/config.yml]: I'm still not convinced it's better to repoint both PRs instead of just repointing one.

File: packages/celotool/src/e2e-tests/governance_tests.ts
Comment (asaj) [inline packages/celotool/src/e2e-tests/governance_tests.ts]: nit: consider `blockBaseGasFee`

File: packages/celotool/src/e2e-tests/transfer_tests.ts
Comment (asaj) [inline packages/celotool/src/e2e-tests/transfer_tests.ts:76]: Was this not used, or..?

File: packages/celotool/src/e2e-tests/transfer_tests.ts
Comment (asaj) [inline packages/celotool/src/e2e-tests/transfer_tests.ts]: speed

File: packages/celotool/src/e2e-tests/transfer_tests.ts
Comment (asaj) [inline packages/celotool/src/e2e-tests/transfer_tests.ts:466]: Why did this change?

File: packages/celotool/src/e2e-tests/transfer_tests.ts
Comment (asaj) [inline packages/celotool/src/e2e-tests/transfer_tests.ts]: 5k or 15k?

Comment (asaj) [inline packages/celotool/src/e2e-tests/transfer_tests.ts]: Can we restart with clean nodes so we don't have this funky check?

**Review by asaj — APPROVED** [review]
Looks great! Mostly nits

File: packages/celotool/src/e2e-tests/transfer_tests.ts
Comment (nategraf) [reply packages/celotool/src/e2e-tests/transfer_tests.ts:76]: It wasn't used.

File: .circleci/config.yml
Comment (nategraf) [reply .circleci/config.yml]: My intention is to merge celo-org/celo-blockchain#563, point this back to master, wait for CI to pass, then merge this one. I simply want to see that the e2e tests pass (or fail) on this PR while the other is still pending.

File: packages/celotool/src/e2e-tests/transfer_tests.ts
Comment (nategraf) [reply packages/celotool/src/e2e-tests/transfer_tests.ts:466]: It was out of date and being ignored. I adjusted each of the gas costs based on master then added a step step to enforce that they are accurate.

File: packages/celotool/src/e2e-tests/transfer_tests.ts
Comment (nategraf) [reply packages/celotool/src/e2e-tests/transfer_tests.ts]: Restarting with clean nodes takes longer and reduces test coverage because it only considers transfers to fresh accounts. I'll go ahead and split this off into it's own test case though.

File: packages/celotool/src/e2e-tests/governance_tests.ts
Comment (nategraf) [reply packages/celotool/src/e2e-tests/governance_tests.ts]: Done

File: packages/celotool/src/e2e-tests/transfer_tests.ts
Comment (nategraf) [reply packages/celotool/src/e2e-tests/transfer_tests.ts]: Done

File: packages/celotool/src/e2e-tests/transfer_tests.ts
Comment (nategraf) [reply packages/celotool/src/e2e-tests/transfer_tests.ts]: I should really leave math to the computers

**Pull request merged**
